knowledge-base/treeview-change-expand-collapse-icons.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Change the Expand and Collapse Icons in TreeView
+description: Learn how to customize the expand and collapse icons in the TreeView component for Blazor.
+type: how-to
+page_title: How to Customize Expand and Collapse Icons in Blazor TreeView
+slug: treeview-kb-change-expand-collapse-icons
+tags: treeview, blazor, icons, expand, collapse, customization
+res_type: kb
+ticketid: 1656109
+---
+
+## Environment
+
+<table>
+<tbody>
+<tr>
+<td>Product</td>
+<td>TreeView for Blazor</td>
+</tr>
+</tbody>
+</table>
+
+## Description
+I want to customize the icons used for expanding and collapsing items in the [TreeView]({%slug treeview-overview%}) for Blazor. 
+
+This knowledge base article answers the following questions:
+- How can I use custom icons for the TreeView expand and collapse functionality?
+- Is it possible to change the default expand and collapse icons in the TreeView for Blazor?
+
+## Solution
+
+You can change the expand/collapse icons in the TreeView by overriding the built-in icons with other icons using [custom CSS rules]({%slug themes-override%}). In addition, you can use the `Class` parameter of the TreeView to add a custom CSS Class and modify a specific instance of the TreeView, instead of all instances on the page.
+
+>caption Change the expand/collapse icons in TreeView
+
+````CSHTML
+@* If your Telerik.UI.for.Blazor version is 4.3.0 or later, use the following CSS. *@
+@* Render the desired SVG icon and inspect it with your dev tools to get its path. *@
+<style>
+    .custom-icons .k-treeview-toggle .k-svg-icon.k-svg-i-caret-alt-down svg path {
+        d: path("m382.059 158.059-126.06 126.06-126.061-126.06L96 192l159.999 160L416 192z");
+    }
+
+    .custom-icons .k-treeview-toggle .k-svg-icon.k-svg-i-caret-alt-right svg path {
+        d: path("m158.059 129.941 126.06 126.06-126.06 126.061L192 416l160-159.999L192 96z");
+</style>
+
+@* If your Telerik.UI.for.Blazor version is below 4.3.0, use this CSS. *@
+@* Copy the unicode of your desired icon from the Progress Design System - https://www.telerik.com/design-system/docs/foundation/iconography/icon-list/ *@
+
+@* <style>
+    .custom-icons .k-treeview-toggle .k-icon.k-i-caret-alt-down::before {
+        content: "\e015";
+    }
+
+    .custom-icons .k-treeview-toggle .k-icon.k-i-caret-alt-right::before {
+        content: "\e014";
+    }
+</style> *@
+
+<TelerikTreeView Class="custom-icons"
+                 Data="@FlatData"
+                 @bind-ExpandedItems="@ExpandedItems" />
+
+@code {
+    private IEnumerable<TreeItem> FlatData { get; set; }
+
+    private IEnumerable<object> ExpandedItems { get; set; } = new List<TreeItem>();
+
+    protected override void OnInitialized()
+    {
+        FlatData = GetFlatData();
+
+        ExpandedItems = FlatData.Where(x => x.HasChildren == true).ToList();
```

Comment:
You can consider expanding only some of the items, so that both custom icons are initially visible without the need to do anything.